### PR TITLE
Fix #510: Async readback deadlock with direct send compositing

### LINKED ISSUE #510: Async readback deadlock with direct send compositing
The transfer window creation from the transfer thread requires a command handler on the pipe thread, which is blocked waiting for the other thread's input frame in the compositor/frameAssemble.

@hernando reported this.

### PR DESCRIPTION
**Pull request by eile**

### CODE REVIEW

**Comment by eile** [comment]
@tribal-tec @hernando : Please review carefully. I think this should not move any operations to the wrong thread, but this qt threading code tends to be fragile.

**Comment by eile** [comment]
NB: It actually deletes the transfer window in the pipe thread, instead of the transfer thread now. To me this is more symmetric.

Shit, just realized that the thread needs to be moved back. Be right back with an amend, feel free to review the create part already.

**Comment by eile** [comment]
NB2: I think #177 did not go far enough and complicated things instead of doing what this PR does now.

**Comment by tribal-tec** [comment]
Could work, but only a test on Windows with Qt will tell you definitely. I remember changing this part because of the Windows test, so I guess I'll have to do one again...

File: eq/pipe.h [hunk 1]
```diff
@@ -190,6 +190,9 @@ class Pipe : public fabric::Pipe< Node, Pipe, eq::Window, PipeVisitor >
     /** @internal Start the async readback thread. */
     bool startTransferThread();
 
+    /** @internal @return the qthread, or nullptr. */
+    QThread* getTransferQThread();
```
Comment (hernando) [inline eq/pipe.h]: #ifdef EQ_QT_USED?

File: eq/channel.cpp [hunk 2]
```diff
@@ -1224,15 +1221,22 @@ void Channel::_finishReadback( const co::ObjectVersion& frameDataVersion,
                                const co::NodeIDs& netNodes )
 {
     LBLOG( LOG_TASKS|LOG_ASSEMBLY ) << "Finish readback" << std::endl;
-    FrameDataPtr frameData = getNode()->getFrameData( frameDataVersion );
-    const Images& images = frameData->getImages();
-    Image* image = images[ imageIndex ];
-    const GLEWContext* glewContext = getWindow()->getTransferGlewContext();
 
+    Window* window = getWindow();
```
Comment (bilgili) [inline eq/channel.cpp]: const

File: eq/pipe.h (hunk 1 above)
Comment (eile) [inline eq/pipe.h]: class QThread; forward declaration in types.h, used as pointer pass-on only.

Comment (hernando) [inline eq/pipe.h]: yep, I saw it later

File: eq/channel.cpp (hunk 2 above)
Comment (eile) [inline eq/channel.cpp]: done

**Comment by hernando** [comment]
I've tested this under Linux with the applications and configuration that was triggering the bug and now it works.

**Comment by eile** [comment]
Soo... +1?

Note additional commit to fix a bug.

**Comment by hernando** [comment]
+1 for me, but just because it works not that I understand the change.

**Pull request merged**